### DISCUSSION TITLE
Programmatically setting values are causing isDirty and isTouched to be set to true.

### DISCUSSION THREAD

**Discussion by asadbekabduvoitov-nordra** [TanStack/form#991] [Ideas]
I'm working on a project, where id will be automatically generated when modal is opened. And my code looks like something like this:

```
useEffect(() => {
	(async () => {
		if (form.state.values.employee_id) return;

		const { data } = await supabase.rpc("generate_eid_for_org_users");
		form.setFieldValue("employee_id", String(data?.split(",")[0]));
	})();
}, []);
```

Where I'm programmatically setting value into input field. But this way of setting value is causing isDirty and isTouched to be set to true, and as a result my global error warning is getting rendered, even without user interacts with the given fields. And I couldn't any find any way to fix the issue. Even I asked from AI:
 
![screenshot_2024-10-22_at_21 38 16_720](https://github.com/user-attachments/assets/da83c61c-d13c-4ce3-9e9e-b656b8605374)

Does framework provide way to set values without update filed states ?
If not it would be better to see this feature in upcoming versions 🙌.

**Comment by jacekwilczynski** [comment]
Hi! Sounds like what you're asking for already exists:
```ts
form.setFieldValue("employee_id", String(data?.split(",")[0]), { dontUpdateMeta: true });
```

**Reply by asadbekabduvoitov-nordra** [reply]
Providing dontUpdateMeta as true, is not setting the value at all in my scenario. I have tested it, but It's not setting the value into the field. But If I set without giving or setting dontUpdateMeta as false it's working in fact.

**Reply by jacekwilczynski** [reply]
That's surprising... Looking at TSF source code I can't see a way this could happen. Just a random idea:

Maybe the value does get updated, but the component that displays it does not re-render? Seems unlikely, but if, by any chance, your component subscribes (with `<form.Subscribe ...` or `form.useStore`) to the field meta but not the field value (and for whatever reason you didn't use `<form.Field ...` or `form.useField`), it would keep displaying a stale value if the meta didn't update.

Probably not your case, but that's the only thing that comes to my mind. Perhaps if I saw a bit more of the code (or, even better, a sandbox reproducing the issue) I could help.
